src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -64,7 +64,7 @@ public abstract partial class Renderer : IDisposable, IAsyncDisposable
     /// <param name="serviceProvider">The <see cref="IServiceProvider"/> to be used when initializing components.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
     public Renderer(IServiceProvider serviceProvider, ILoggerFactory loggerFactory)
-        : this(serviceProvider, loggerFactory, GetComponentActivatorOrDefault(serviceProvider))
+        : this(serviceProvider, loggerFactory, GetComponentActivatorOrDefault(serviceProvider)!)
```

Comment:
It didn't seem necessary to change the signature of the other overload to make `IComponentActivator` nullable. Let me know if you feel otherwise.

Reply:
I haven't really 100% internalised why it's desirable to remove `DefaultComponentActivator` and bake the default activation logic into `ComponentFactory` (instead of continuing to have the default logic inside `DefaultComponentActivator`). Perhaps this approach lets you avoid some code duplication, I'm not sure.

If we're going with the approach shown here, I think we should at least add a comment saying `Even though IComponentActivator is not marked as nullable, we do allow null because that's how the framework internally indicates that we should use default activation logic`.

Reply:
> instead of continuing to have the default logic inside DefaultComponentActivator

We have to cache two related, but distinct things (object factory + property initializer) while avoiding the penalty of an additional lookup in the default case. Today the object initialization is delegated to the component intiailizer, and the property intiailization to the factory. Avoiding two looks up meant either special casing the `DefaultComponentActivator`  within the factory or smooshing together everything in the factory. The latter seemed a little nicer since it also meant avoiding an extra allocation (per circuit).